lab_2_tokenize_by_bpe/main.py

```diff
@@ -26,6 +44,23 @@ def collect_frequencies(
     :param end_of_word: a token that signifies the end of word
     :return: dictionary in the form of <preprocessed word: number of occurrences>
     """
+    if not (isinstance(text, str)
+            and isinstance(start_of_word, (str, type(None)))
+            and isinstance(end_of_word, str)
+    ):
+        return None
+
+    frequencies = {}
+    words = text.split()
+
+    for word in words:
+        prepared_word = prepare_word(word, start_of_word, end_of_word)
+        if prepared_word is None:
+            return None
+
+        frequencies.update({prepared_word: words.count(word)})
```

Comment:
`update` exists to add multiple keys in the dictionary at once, if you only need to add one key, consider adding it traditionally with the square brackets